test/malloc_compliance_tests_glibc.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2023 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// run basic compliance tests for glibc's functions (sanity check)
+
+#include "base.hpp"
+
+using umf_test::test;
+
+#include <assert.h>
+#include <stdlib.h>
+
+#include "../src/memory_pool_internal.h"
+#include "malloc_compliance_tests.hpp"
+#include <umf/memory_pool_ops.h>
+
+// casting to (uintptr_t) because of:
+// warning C4312: 'type cast': conversion from 'unsigned int' to 'void *' of greater size
+#define POOL_PRIV ((void *)(uintptr_t)0xDEADBEAF)
+
+static void *glibc_malloc(void *pool, size_t size) {
+    (void)pool; // unused
+    assert(pool == POOL_PRIV);
+    return malloc(size);
+}
+
+static void *glibc_calloc(void *pool, size_t num, size_t size) {
+    (void)pool; // unused
+    assert(pool == POOL_PRIV);
+    return calloc(num, size);
+}
+
+static void *glibc_realloc(void *pool, void *ptr, size_t size) {
+    (void)pool; // unused
+    assert(pool == POOL_PRIV);
+    return realloc(ptr, size);
+}
+
+static enum umf_result_t glibc_free(void *pool, void *ptr) {
+    (void)pool; // unused
+    assert(pool == POOL_PRIV);
+    free(ptr);
+    return UMF_RESULT_SUCCESS;
+}
+
+static const umf_memory_pool_ops_t UMF_GLIBC_POOL_OPS = {
+    /* .version = */ UMF_VERSION_CURRENT,
+    /* .initialize = */ NULL,
+    /* .finalize = */ NULL,
+    /* .malloc = */ glibc_malloc,
+    /* .calloc = */ glibc_calloc,
+    /* .realloc = */ glibc_realloc,
+    /* .aligned_malloc = */ NULL,
+    /* .malloc_usable_size = */ NULL,
+    /* .free = */ glibc_free,
+    /* .get_last_allocation_error = */ NULL,
+};
+
+static umf_memory_pool_t UMF_GLIBC_POOL = {
+    /* .pool_priv = */ POOL_PRIV,
+    /* .ops = */ UMF_GLIBC_POOL_OPS,
+    /* .provider = */ NULL,
+};
+
+TEST_F(test, glibc_malloc_compliance_test) {
```

Comment:
Why not `umfPoolTest `? With `umfPoolTest` and TEST_P instead of TEST_F it would be possible to run those testcases for multiple pool types, without having to manually instantiate each case for each pool. See here as an example: https://github.com/oneapi-src/unified-memory-framework/blob/main/test/disjoint_pool.cpp#L163

Reply:
Why is the current solution wrong? Would the one with `umfPoolTest` and `TEST_P` be simpler than this?
```c
TEST_F(test, glibc_malloc_compliance_test) {
    malloc_compliance_test(&UMF_GLIBC_POOL);
}

TEST_F(test, glibc_calloc_compliance_test) {
    calloc_compliance_test(&UMF_GLIBC_POOL);
}

TEST_F(test, glibc_realloc_compliance_test) {
    realloc_compliance_test(&UMF_GLIBC_POOL);
}

TEST_F(test, glibc_free_compliance_test) {
    free_compliance_test(&UMF_GLIBC_POOL);
}
```

Reply:
`TEST_P` gives you the ability to parametrize your tests, so instead of hardcoding umf_glibc_pool, you could use a collection of all the pool implementations as defined by the `umfPoolTest` fixture. So you'd only have one malloc test, not one for each possible implementation.

Reply:
@ldorau please run these tests on the disjoint pool also and use TEST_P in this case. I think then we could compare both approaches and choose one.

Reply:
Done